content/docs/start/model-management/model-cicd.md

```diff
@@ -0,0 +1,179 @@
+      uses: iterative/gto-action@v2
+  outputs:
+    event: ${{ steps.gto.outputs.event }}
+    name: ${{ steps.gto.outputs.name }}
+    stage: ${{ steps.gto.outputs.stage }}
+    version: ${{ steps.gto.outputs.version }}
+```
+
+The next job called "test-model-artifact" actually performs the action. First,
+it uses the outputs of the parse job and checks whether the action should be
+performed. If the tag was produced by the model registry and if the
+corresponding action was assignment to the "test" stage, it proceeds with the
+rest of the workflow.
+
+```yaml
+test-model-artifact:
+  needs: parse
+  if:
+    "${{ needs.parse.outputs.event == 'assignment' }} && ${{
+    needs.parse.outputs.stage == 'test' }}"
+  runs-on: ubuntu-latest
+```
+
+After the next steps which set up Python and instal necessary requirements, the
+job adds our project's DVC remote so that the workflow can download the model
+without having an explicit URL
+
+```yaml
+- name: Add DVC Remote
+      run:  dvc remote add -d --local storage s3://dvc-public/remote/get-started-pools
+```
+
+Then in the last step of the job, we use `dvc get` to download the model
+similarly to how we did it
+[above](/docs/start/model-management/model-cicd#downloading-models), except now
+we don't need to add an explicit URL (this way, the workflow will keep working
+even if we rename our git repository).
+
+Finally, the job just checks the model size and fails if the size is too large.
+
+You can use the following template to create your own CICD actions on GitHub!
+
+```yaml
+name: Template CICD Action for DVC Model Registry
+
+on:
+  # the workflow is triggered whenever a tag is pushed to the repository
+  push:
+    tags:
+      - "*"
+
+jobs:
+
+  # This job parses the git tag with the GTO GitHub Action to identify model registry actions
+  parse:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+    - name: "Parse GTO tag"
+      id: gto
+      uses: iterative/gto-action@v2
+    outputs:
+      event: ${{ steps.gto.outputs.event }}
+      name: ${{ steps.gto.outputs.name }}
+      stage: ${{ steps.gto.outputs.stage }}
+      version: ${{ steps.gto.outputs.version }}
+
+  perform-action:
+    needs: parse
+    if: # here paste the model registry event condition you want to trigger your action
+    steps:
+
+    ...
+
+```
+
+<admon type="tip" id="sagemaker-and-gitlab">
+
+The same ingredients we will use to set up this workflow are be usable for any
+other CICD action. You can check out a
+[similar workflow in our example repository](https://github.com/iterative/example-get-started-experiments/blob/main/.github/workflows/deploy-model.yml)
+which deploys a specific version of the model to an Amazon Sagemaker endpoint
+for inference whenever it is assigned to a stage.
+
+With other git servers such as GitLab you can use the same logic (you will only
+have to adapt the syntax and use the GTO library manually to parse tags with the
```

Comment:
@dberenbaum I will change this tip once I rewrite the above without the GTO action...in fact I would replace it with something like "on GitHub you can make things even simpler with our GTO GitHub action...

Reply:
I am wondering whether to actually remove the GTO action from the guide. We should show other options either way but this is the simplest way to do things.

So maybe we can have a separate user guide where we show how it is done without the GTO action and also how it is done on GitLab (and potentially BitBucket)? 

This is refering to GitHub anyway so you will need to modify the get started guide for GitLab even if we used GTO manually. So it makes more sense to me to link to those other options instead and keep this tutorial slightly opaque but simpler.

WDYT @dberenbaum ?

Reply:
I don't know how complex it looks without the action, but I'm fine to leave it and discuss as a follow-up. In that case, should we move this info up to the area where we include the gto action?

Reply:
yes, agreed